.github/workflows/build-and-test-orchard-core.yml

```
@@ -213,6 +213,10 @@ jobs:
           binary-log-artifact-retention-days: ${{ inputs.build-binary-log-artifact-retention-days }}
           warnings-as-errors: ${{ inputs.warnings-as-errors }}
 
+      - name: Print configuration summary
+        if: always()
```

Do we need this report if the workflow is canceled?
If not:
```suggestion
        if: success() || failure()
```

---

Yes, we want the report most of all in the case of an error 😁

---

The `always()` condition is evaluated as `true` in case of manual cancellation or cancellation by another workflow while `success() || failure()` is not.
See related: https://lombiq.atlassian.net/wiki/spaces/DEV/pages/2545614855/Writing+GitHub+Actions+workflows+and+actions#Conventions

---

I am aware of that, and like a said, we want this info to show up **specifically** when a workflow job fails (which then cancels it).

---

You say, we need this information if eg the [spelling](https://github.com/Lombiq/GitHub-Actions/blob/dev/.github/workflows/spelling.yml) workflow failed?

I'm sure, the `Print configuration summary` step will be executed in case the `Build and Static Code Analysis` get failed if you are using `success() || failure()` condition here.
See condition in [Cancel Workflow on Failure](https://github.com/Lombiq/GitHub-Actions/blob/30aacfb73e7bb869cf83063547d0344ff0ae44e8/.github/workflows/build-and-test-orchard-core.yml#L240)

https://github.com/Lombiq/GitHub-Actions/blob/30aacfb73e7bb869cf83063547d0344ff0ae44e8/.github/workflows/build-and-test-orchard-core.yml#L239-L243